### DISCUSSION TITLE
Jwt authentication in next js

### DISCUSSION THREAD

**Discussion by Sandeep06Dev** [vercel/next.js#15600] [Help]
Hi brilliant guys.. 

I want any code snippets by which we can add jwt authentication in few pages.

E.g

I have a login.jsx and profile.jsx.

I want to open profile.jsx only when its has a valid jwt token.. Additional if any one try to open profile.jsx from address bar (without jwt authentication) then they 
should be redirect to login.jsx

**Comment by enochndika** [comment]
Check this [sandbox](https://codesandbox.io/s/vibrant-ride-p2rr5), i have used context api to authenticated on client side

**Reply by Dannymx** [reply]
The link doesn't seem to be working.

**Reply by enochndika** [reply]
i have created a new [sandbox](https://codesandbox.io/s/stupefied-paper-28h1f)

**Reply by Dannymx** [reply]
Thanks for the link.

I think it would be nice if the team from Next.js could create an example using SSG where the frontend requests JWT (with emai/password or other login method) then how to handle the callback/save the jwt in a cookie.

**Reply by mcp-net** [reply]
your sandbox doesn't have all needed component.

You are missing everything under "/api" folder, like "/api/user", "/api/user/[id]

Check your sandbox environment.

**Reply by enochndika** [reply]
no nothing is missing, just use your own backend, my example is just an illustration.  in your backend you will need the token and the user id
